docker-compose.yml

```diff
@@ -0,0 +1,210 @@
+    container_name: oppia-cloud-datastore
+    ports:
+      - "8089:8089"
+    environment:
+      - CLOUDSDK_CORE_DISABLE_PROMPTS=1
+    volumes:
+      - cloud_datastore_emulator_cache:/app/cloud_datastore_emulator_cache
+    healthcheck:
+      test: curl -f http://localhost:8089/ || exit 1
+      timeout: 10s
+      interval: 2m
+      retries: 5
+      start_period: 7m
+    entrypoint: /google-cloud-sdk/bin/gcloud
+    command:
+      - "beta"
+      - "emulators"
+      - "datastore"
+      - "start"
+      - "--project"
+      - "dev-project-id"
+      - "--data-dir"
+      - "/app/cloud_datastore_emulator_cache"
+      - "--host-port=0.0.0.0:8089"
+      - "--consistency=1.0"
+      - "--quiet"
+      - "--no-store-on-disk"
+
+  firebase:
+    image: node:16.13.0-alpine
+    container_name: oppia-firebase-emulator
+    ports:
+      - "9099:9099"
+      - "4000:4000"
+    volumes:
+      - ./.firebase.json:/app/oppia/.firebase.json
+      - shared_node_modules:/app/oppia/node_modules
+      - firebase_emulator_cache:/app/firebase_emulator_cache
+    # NOTE: The firebase service depends on the packages from node_modules that are being
+    # installed in the Frontend Stage of Dockerfile. In future, we can remove the dependency
+    # on Frontend Stage and install the required packages in this service itself.
+    depends_on:
+      angular-build:
+        condition: service_healthy
+      webpack-compiler:
+        condition: service_healthy
+    healthcheck:
+      test: wget --spider http://localhost:4000/auth/ && echo 'y' | /app/oppia/node_modules/firebase-tools/lib/bin/firebase.js emulators:export /app/firebase_emulator_cache --project=dev-project-id || exit 1
+      timeout: 20s
+      interval: 20s
+      retries: 5
+    entrypoint:
+      - /app/oppia/node_modules/firebase-tools/lib/bin/firebase.js
+      - "emulators:start"
+      - "--only"
+      - "auth"
+      - "--project=dev-project-id"
+      - "--config=/app/oppia/.firebase.json"
+    command:
+      - "--import=/app/firebase_emulator_cache"
+      - "--export-on-exit=/app/firebase_emulator_cache"
+
+  redis:
+    image: redis:6.2.4
+    container_name: oppia-redis
+    ports:
+      - "6379:6379"
+    environment:
+      - PATH=/usr/local/bin/
+    volumes:
+      - redis_dump:/data/dump.db
+      - ./redis_docker.conf:/app/oppia/redis_docker.conf
+    healthcheck:
+      test: redis-cli ping
+      timeout: 5s
+      interval: 30s
+      retries: 5
+    command: redis-server /app/oppia/redis_docker.conf
+
+  elasticsearch:
+    image: elasticsearch:7.17.0
+    container_name: oppia-elasticsearch
+    ports:
+      - "9200:9200"
+    environment:
+      - ES_JAVA_OPTS=-Xms100m -Xmx500m
+      - ES_PATH_CONF=/usr/share/elasticsearch/config
+      - discovery.type=single-node
+    healthcheck:
+      test: curl -f http://localhost:9200/ || exit 1
+      timeout: 10s
+      interval: 1m
+      retries: 5
+      start_period: 5m
+    command: elasticsearch -q
+
```

Comment:
Optional: Create a network with the name `backend` and attach all backend services to that network.

Reply:
added in #18638